proof/file.go

```diff
@@ -172,6 +197,13 @@ func (f *File) Decode(r io.Reader) error {
 		return err
 	}
 
+	// Cap the number of proofs there can be within a single file to avoid
+	// OOM attacks. See the comment for FileMaxNumProofs for the reasoning
+	// behind the value chosen.
+	if numProofs > FileMaxNumProofs {
+		return ErrProofFileInvalid
+	}
```

Comment:
Might be a good idea to add context here that the error refers to the number of proofs and not the size of the proof file. Or a specific error type. Not a blocker though, IMO.